### ISSUE TITLE
reflec-ts status?

### ISSUE THREAD

**Issue by maryo** [pcan/reflec-ts-example#1]
Hi. I am very interested in the reflec-ts. Still have plans with the project? The result looks promising. I guess it's private since i could'nt find any source code.

**Comment by pcan** [comment]
Hi, you're right, the reflec-ts project is still private but for a few simple reasons: one is the list of [open points](https://github.com/pcan/reflec-ts-example#open-points-tbd) which involves the core of Typescript compiler. Since I don't want heavy coupling with the compiler code and I don't want to release a custom version for each new official Typescript release I decided to not publish the reflec-ts pre-processor. I think this is a wise choice, since if you or other developers start using reflec-ts I want to be sure that it can evolve without having to tweak our development environment in order to use a custom compiler. If you know the [Atom Editor](https://atom.io), for example, it has a very good integration with Typescript and it allows the choice of a custom compiler version. This may not be true for all IDEs. I wish to have an API-like integration with the compiler: this should happen in a close future, after [this issue](https://github.com/Microsoft/TypeScript/issues/5595) is merged in the 2.0 version of the compiler. Furthermore, there is another [open discussion](https://github.com/Microsoft/TypeScript/issues/3628) in which I'm involved, that may evolve in a good way for the reflect-ts project.

TL;DR I prefer to wait some new features in the official compiler that allow a smoother integration with reflec-ts, but **YES, this project is still alive**.

**Comment by maryo** [comment]
Thanks for the info. That's great it is still alive. I had a simmilar idea in mind and that's why I'm asking.

I see the reasons. But ...maybe if the amount of the required changes in the compiler was limited, would it not be an acceptable temporary solution? I use my own TS fork too. Maybe if the patch was automated it would be possible to minimize the possible future breaks? After all, the releases are not so often and I'm afraid they will solve most things but I guess it won't be this year. But maybe it will, who knows.

Anyway I'll wait and wish you good luck :)

**Comment by stepancar** [comment]
pcan, could you help, please? https://github.com/Microsoft/TypeScript-wiki/issues/65

**Comment by monobyte** [comment]
Any news here? If you're struggling with time, why not commit what you've got so others can contribute?

**Comment by pcan** [comment]
I will commit the code as soon as possible, I still have to do some improvements. Please consider that with the 2.0 they replaced the emitter with an AST transformer, which makes things really simpler. Currently, the reflect-ts code is still bound to the 1.7 emitter code and it makes no sense to release it now as is.

**Comment by pcan** [comment]
reflec-ts is finally [here](https://github.com/pcan/reflec-ts).
